hw/ip/csrng/data/csrng_testplan.hjson

```diff
@@ -13,55 +13,78 @@
       desc: '''
-            Verify recov_alert asserts as predicted.
-            Verify all recov_alert_sts bits assert/clear as predicted.
-            Verify fatal_alert asserts as predicted.
+            Verify recov_alert asserts when recov_alert_sts becomes non-zero.
+            Verify each respective bit of recov_alert_sts asserts when:
+            - Either of the 3 values in the control register are not valid kMultiBitBool values.
+            - When an initiate or reseed command is sent where flag0 is not a valid kMultiBitBool
+            value.
+            - The genbits bus value is equal to the prior valid value.
+            - When an illegal command is used (0x0,0x6-0xf).
+            Verify that writing zeros to the recoverable alert status register resets all the
+            status bits.
+            Verify fatal_alert asserts when:
+            - An illegal state is reached.
+            - The AES block raises a fatal alert.
+            - There is an integrity failure on the bus.
             '''
       stage: V2
       tests: ["csrng_alert"]
     }
     {
       name: err
       desc: '''
-            Verify err_code register bits assert/clear as predicted.
+            Verify err_code register bits assert when:
+            - An error has been detected in any of the internal fifos and the corresponding
+            write/read/state bits.
+            - An illegal state is reached in any of the 6 state machines.
+            Verify that the err_code register clears all bits after reset.
             '''
       stage: V2
       tests: ["csrng_err"]
     }
     {
       name: cmds
       desc: '''
-            Verify all SW app csrng commands req/status behave as predicted.
-            Verify all HW app csrng commands req/status behave as predicted.
+            Verify all csrng commands req/status behave as predicted on all applications: HW0, HW1
+            and SW.
             Verify above for all valid values of acmd, clen, flags, glen.
             Verify for multiple hw app interfaces running in parallel.
             Verify sw/hw app interfaces running in parallel.
-            Verify internal state for sw/hw apps.
-            Verify genbits generated as predicted.
-            Verify fips bit is passed through properly.
-            Verify ability to access registers based on otp, enables.
-            Verify AES_HALT.
+            Verify main_sm_state for sw/hw apps.
+            Verify that genbits generates the amount specified by glen.
+            Verify fips bit is set to the inverse of flag0 that was used on the last initialize or
+            reseed.
+            Verify that if otp_en_csrng_sw_app_read is set, genbits and int_state_value registers
+            are readable, and unreadable otherwise.
+            Verify that when AES_HALT is set during a generate command that no request is sent to
+            the AES block.
             Verify commands with continuous/non-continuous valid.
```

Comment:
We should also verify that if the FIPS bit drops at the input (from entropy_src) the generated block produced from this entropy input will also have the FIPS bit low. This is another TODO.